zetaclient/inbound_tracker.go

```diff
@@ -0,0 +1,263 @@
+	hash, err := chainhash.NewHashFromStr(txHash)
+	if err != nil {
+		return "", err
+	}
+	tx, err := ob.rpcClient.GetRawTransactionVerbose(hash)
+	if err != nil {
+		return "", err
+	}
+	blockHash, err := chainhash.NewHashFromStr(tx.BlockHash)
+	if err != nil {
+		return "", err
+	}
+	block, err := ob.rpcClient.GetBlockVerbose(blockHash)
+	if err != nil {
+		return "", err
+	}
+	tss, err := ob.zetaClient.GetBtcTssAddress()
+	if err != nil {
+		return "", err
+	}
+	event, err := GetBtcEvent(*tx, tss, uint64(block.Height), &ob.logger.WatchInTx)
+	if err != nil {
+		return "", err
+	}
+	if event == nil {
+		return "", errors.New("no btc deposit event found")
+	}
+	msg := ob.GetInboundVoteMessageFromBtcEvent(event)
+	if !vote {
+		return msg.Digest(), nil
+	}
+	zetaHash, err := ob.zetaClient.PostSend(PostSendEVMGasLimit, msg)
+	if err != nil {
+		ob.logger.WatchInTx.Error().Err(err).Msg("error posting to zeta core")
+		return "", err
+	}
+	ob.logger.WatchInTx.Info().Msgf("ZetaSent event detected and reported: PostSend zeta tx: %s", zetaHash)
+	return msg.Digest(), nil
+}
+
+func (ob *EVMChainClient) ObserveTrackerSuggestions() error {
+	trackers, err := ob.zetaClient.GetInboundTrackersForChain(ob.chain.ChainId)
+	if err != nil {
+		return err
+	}
+	for _, tracker := range trackers {
+		ob.logger.ExternalChainWatcher.Info().Msgf("checking tracker with hash :%s and coin-type :%s ", tracker.TxHash, tracker.CoinType)
+		switch tracker.CoinType {
+		case common.CoinType_Zeta:
+			ballotIdentifier, err := ob.CheckReceiptForCoinTypeZeta(tracker.TxHash, true)
+			if err != nil {
+				return err
+			}
+			ob.logger.ExternalChainWatcher.Info().Msgf("Vote submitted for inbound Tracker,Chain : %s,Ballot Identifier : %s, coin-type %s", ob.chain.ChainName, ballotIdentifier, common.CoinType_Zeta.String())
+		case common.CoinType_ERC20:
+			ballotIdentifier, err := ob.CheckReceiptForCoinTypeERC20(tracker.TxHash, true)
+			if err != nil {
+				return err
+			}
+			ob.logger.ExternalChainWatcher.Info().Msgf("Vote submitted for inbound Tracker,Chain : %s,Ballot Identifier : %s, coin-type %s", ob.chain.ChainName, ballotIdentifier, common.CoinType_ERC20.String())
+		case common.CoinType_Gas:
+			ballotIdentifier, err := ob.CheckReceiptForCoinTypeGas(tracker.TxHash, true)
+			if err != nil {
+				return err
+			}
+			ob.logger.ExternalChainWatcher.Info().Msgf("Vote submitted for inbound Tracker,Chain : %s,Ballot Identifier : %s, coin-type %s", ob.chain.ChainName, ballotIdentifier, common.CoinType_Gas.String())
+		}
+	}
+	return nil
+}
+
+func (ob *EVMChainClient) CheckReceiptForCoinTypeZeta(txHash string, vote bool) (string, error) {
+	connector, err := ob.GetConnectorContract()
+	if err != nil {
+		return "", err
+	}
+	hash := ethcommon.HexToHash(txHash)
+	receipt, err := ob.evmClient.TransactionReceipt(context.Background(), hash)
+	if err != nil {
+		return "", err
+	}
+
+	logValues := make([]ethtypes.Log, len(receipt.Logs))
+	for i, log := range receipt.Logs {
```

Comment:
Would it be simpler to iterate directly on `receipt.Logs` instead of a copied `logValues`?